File: sangjin/level1/29_약수의_합.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    vector<int> sum(3001, 0);

    for(int i = 1; i < sum.size(); i++){
        for(int j = 1; i * j <= sum.size() - 1; j++){
            sum[i*j] += j;
        }
    }

    return sum[n];
}

File: yunj/level1/29_약수의 합.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int solution(int n) {
        int answer = 0;
        Set<Integer> s = new HashSet<>();
        for (int i = 1; i <= n; i++) {
            if (n % i == 0) {
                s.add(i);
            }
        }
        for (int num : s) {
            answer += num;
        }
        return answer;
    }
}